activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/process/BpmnModelServiceTaskCatchBoundaryValidator.java

```diff
@@ -0,0 +1,91 @@
+
+import static java.lang.String.format;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.activiti.bpmn.model.BoundaryEvent;
+import org.activiti.bpmn.model.BpmnModel;
+import org.activiti.bpmn.model.ErrorEventDefinition;
+import org.activiti.bpmn.model.ServiceTask;
+import org.activiti.cloud.modeling.api.ModelValidationError;
+import org.activiti.cloud.modeling.api.ValidationContext;
+
+/**
+ * Implementation of {@link BpmnCommonModelValidator} for validating service task boundaries
+ */
+public class BpmnModelServiceTaskCatchBoundaryValidator implements BpmnCommonModelValidator {
+
+    public static final String MISSING_BOUNDARY_WARNING = "Missing Catch Error boundary event";
+    public static final String INVALID_SERVICE_IMPLEMENTATION_DESCRIPTION = "The service implementation on service '%s' might fail silently";
+    public static final String SERVICE_TASK_VALIDATOR_NAME = "BPMN service task catch boundary validator";
+
+    private final FlowElementsExtractor flowElementsExtractor;
+
+    private final ServiceTaskImplementationType[] serviceTaskImplementationTypes;
+
+    public BpmnModelServiceTaskCatchBoundaryValidator(FlowElementsExtractor flowElementsExtractor,
+        ServiceTaskImplementationType[] serviceTaskImplementationTypes) {
+        this.flowElementsExtractor = flowElementsExtractor;
+        this.serviceTaskImplementationTypes = Optional.ofNullable(serviceTaskImplementationTypes)
+            .orElse(new ServiceTaskImplementationType[0]);
+    }
+
+    @Override
+    public Stream<ModelValidationError> validate(BpmnModel bpmnModel,
+        ValidationContext validationContext) {
+
+        return flowElementsExtractor.extractFlowElements(bpmnModel, ServiceTask.class).stream()
+            .filter(serviceTask -> serviceTask.getImplementation() != null)
+            .map(serviceTask -> validateServiceTaskBoundary(serviceTask))
+            .filter(Optional::isPresent)
+            .map(Optional::get);
+    }
+
+    private Optional<ModelValidationError> validateServiceTaskBoundary(ServiceTask serviceTask) {
+        if (requiresBoundary(serviceTask)) {
+            return Optional.of(
+                new ModelValidationError(MISSING_BOUNDARY_WARNING,
+                    format(INVALID_SERVICE_IMPLEMENTATION_DESCRIPTION,
+                        serviceTask.getId()), SERVICE_TASK_VALIDATOR_NAME, true));
+        }
+
+        return Optional.<ModelValidationError>empty();
+    }
+
+    private boolean requiresBoundary(ServiceTask serviceTask) {
+        return Arrays.stream(serviceTaskImplementationTypes)
+            .anyMatch(serviceImplementation ->
+                serviceTask.getImplementation().startsWith(serviceImplementation.getPrefix()) &&
+                    (
+                        serviceTask.getBoundaryEvents() == null ||
+                            !hasBoundaryErrorEvent(serviceTask.getBoundaryEvents())
```

Comment:
This would require another PR in activiti core, but it might be worth to add a method `hasBoundaryErrorEvents` (including the null check) directly in the class [Activity](https://github.com/Activiti/Activiti/blob/107acb9ed6a0589d379f63f5c57d913646156f7f/activiti-core/activiti-bpmn-model/src/main/java/org/activiti/bpmn/model/Activity.java#L21). In this way, here you can simply call `serviceTask.hasBoundaryErrorEvents()` and the code will be available if we need the same logic in other places.

Reply:
I think it is a good idea, here the PR: Activiti/Activiti#4197